src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpgradeReadyCheck.cs

```diff
@@ -16,12 +16,17 @@ public interface IUpgradeReadyCheck
         /// </summary>
         string Id { get; }
 
+        /// <summary>
+        /// Gets a message that tells users what they can do if their project is not ready to upgrade.
+        /// </summary>
+        string UpgradeGuidance { get; }
```

Comment:
Would having access to the project be helpful for better diagnostics? If so, the return of `IsReadyAsync` could be more of a record like the UpgradeStepStatus stuff

Reply:
Probably, I almost injected some awareness to say that a feature is not available on .NET5 or .NET6 depending on target.

Instead, I choose a message that says it was only available on .NET full framework. Future versions will probably need more awareness to create better messages.